src/xunit.v3.core/Sdk/v3/Utility/TestCollectionFactoryHelper.cs

```diff
@@ -26,7 +27,8 @@ public static class TestCollectionFactoryHelper
 				.Select(type => new { Type = type, Attribute = type.GetCustomAttributes(typeof(CollectionDefinitionAttribute).AssemblyQualifiedName!).FirstOrDefault() })
 				.Where(list => list.Attribute is not null)
 				.GroupBy(
-					list => list.Attribute!.GetConstructorArguments().Cast<string>().Single(),
+					list => list.Attribute!.GetConstructorArguments().Cast<string>().SingleOrDefault()
```

Comment:
You won't need this once we remove the empty constructor, though the documentation should be updated to indicate that it will only discover test collections that are intended to by discovered by name rather than by type.